sql/catalyst/src/main/scala/org/apache/spark/sql/connector/expressions/expressions.scala

```diff
@@ -388,12 +388,13 @@ private[sql] object HoursTransform {
 }
 
 private[sql] final case class LiteralValue[T](value: T, dataType: DataType) extends Literal[T] {
-  override def toString: String = {
-    if (dataType.isInstanceOf[StringType]) {
-      s"'${StringUtils.replace(s"$value", "'", "''")}'"
-    } else {
-      s"$value"
-    }
+  override def toString: String = dataType match {
+    case StringType => s"'${StringUtils.replace(s"$value", "'", "''")}'"
+    case BinaryType =>
```

Comment:
From a cursory look, it seems so but would be great to check if this complies ANSI, and mention it in PR description.